azure/durable_functions/models/actions/CallSubOrchestratorAction.py

```diff
@@ -10,10 +10,11 @@
 class CallSubOrchestratorAction(Action):
     """Defines the structure of the Call SubOrchestrator object."""
 
-    def __init__(self, function_name: str, _input: Optional[Any] = None, instance_id: str = ""):
+    def __init__(self, function_name: str, _input: Optional[Any] = None,
+                 instance_id: Optional[str] = None):
         self.function_name: str = function_name
         self._input: str = dumps(_input, default=_serialize_custom_object)
-        self.instance_id: str = instance_id
+        self.instance_id: Optional[str] = instance_id
 
         if not self.function_name:
             raise ValueError("function_name cannot be empty")
```

Comment:
Does it make sense to clarify the error message to say `None` or empty, or can we rely on the fact that customers will respect the type annotations?

Reply:
I was hoping we would trust customers would respect type annotations. I think there are also ways to automatically equip the SDK with exceptions if customers do not abide by the type hints, and maybe that's something to add in the future.

If we don't trust the customers to pass the right types, we would need to sprinkle so many type checks everywhere. My perspective here is that Python is very unsafe and so you can't protect customers from everything, especially if type hints are available.

That being said, I'm 100% for adding automatically exception throwing when types are not respective in a future PR

Reply:
Sounds like a reasonable compromise to me.

I know this is a problem in Node, when customers aren't necessarily using TypesScript or even JavaScript tooling that supports typed intellisence.